### PR TITLE
Rename back multipart objects iff read/write Quorum was unavailable

### PR DESCRIPTION
**Pull request by krisis**

### CODE REVIEW

File: xl-v1-object.go [hunk 1]
```diff
@@ -176,7 +176,18 @@ func (xl xlObjects) renameObject(srcBucket, srcObject, dstBucket, dstObject stri
 		if errCount <= len(xl.storageDisks)-xl.readQuorum {
 			return nil
 		}
-		xl.deleteObject(srcBucket, srcObject)
+		// Rename back the object on disks where RenameFile succeeded
+		for index, disk := range xl.storageDisks {
```
Comment (harshavardhana) [inline xl-v1-object.go:180]: Commit has a typo 'iff' should be 'if'

File: xl-v1-object.go [hunk 2]
```diff
@@ -176,7 +176,18 @@ func (xl xlObjects) renameObject(srcBucket, srcObject, dstBucket, dstObject stri
...
+			wg.Add(1)
+			go func(index int, disk StorageAPI) {
+				defer wg.Done()
+				if errs[index] != nil {
```
Comment (harshavardhana) [inline xl-v1-object.go:185]: Routine needs a comment

File: xl-v1-object.go (hunk 1 above)
Comment (krisis) [inline xl-v1-object.go:180]: ["iff"](https://en.wikipedia.org/wiki/If_and_only_if) is not a typo, it expands to "if and only if" in mathematical text. Let me know if I should change it.

**Comment by harshavardhana** [comment]
> "iff" is not a typo, it expands to "if and only if" in mathematical text. Let me know if I should change it.

Better to keep it close to vocabulary.  Tautology perhaps when we do haskell in future :-)

**Pull request merged**